tracer/src/Datadog.Trace/AppSec/WafEncoding/Encoder.cs

```diff
@@ -635,23 +635,19 @@ private static StringBuilder FormatDictionary(IEnumerable<KeyValuePair<string, o
         private static StringBuilder FormatList<T>(IEnumerable<T> objs, StringBuilder sb)
         {
             sb.Append("[ ");
-            using var enumerator = objs.GetEnumerator();
-            if (!enumerator.MoveNext())
-            {
-                sb.Append(" ]");
-                return sb;
-            }
 
-            if (enumerator.Current != null)
+            using var enumerator = objs.GetEnumerator();
+            if (enumerator.MoveNext() && enumerator.Current != null)
             {
                 FormatArgsInternal(enumerator.Current, sb);
-            }
 
-            while (enumerator.MoveNext())
-            {
-                if (enumerator.Current != null)
+                while (enumerator.MoveNext())
```

Comment:
NIT maybe instead which would allow removing some duplicated codes, could be:
```suggestion
                var canMoveNext = enumerator.MoveNext();
            while (canMoveNext)
            {
                if (enumerator.Current != null)
                {
                    FormatArgsInternal(enumerator.Current, sb);
                    canMoveNext = enumerator.MoveNext();
                    if(canMoveNext)
                    {
                        sb.Append(", ");
                    }
                }
            }
```

Reply:
also there's also a `LegacyEncoder` that's being used by default

Reply:
I have updated both. Thanks!